thorlcr/slave/thslavemain.cpp

```diff
@@ -106,27 +106,36 @@ static bool RegisterSelf(SocketEndpoint &masterEp)
         ep.port = getFixedPort(getMasterPortBase(), TPORT_mp);
         Owned<INode> masterNode = createINode(ep);
         CMessageBuffer msg;
+        msg.append(mySlaveNum);
+        queryWorldCommunicator().send(msg, masterNode, MPTAG_THORREGISTRATION);
         if (!queryWorldCommunicator().recv(msg, masterNode, MPTAG_THORREGISTRATION))
             return false;
         PROGLOG("Initialization received");
         unsigned vmajor, vminor;
         msg.read(vmajor);
         msg.read(vminor);
-        if (vmajor != THOR_VERSION_MAJOR || vminor != THOR_VERSION_MINOR)
-        {
-            replyError(TE_FailedToRegisterSlave, "Thor master/slave version mismatch");
-            return false;
-        }
-        Owned<IGroup> rawGroup = deserializeIGroup(msg);
+        Owned<IGroup> processGroup = deserializeIGroup(msg);
+        mySlaveNum = (unsigned)processGroup->rank(queryMyNode());
```

Comment:
>should this be verifying that it matches the slave number that was passed to the master? can it be different?

It does on line 122-123

>It might be cleaner to read config slaveNum earlier, and pass that to the master, rather than initialising mySlaveNum before calling RegisterSelf()

It does when known (in non-container world), then verifies matches rank (lines 122-123)